app/models/computacenter/user_change_generator.rb

```diff
@@ -1,194 +1,81 @@
 class Computacenter::UserChangeGenerator
-  attr_reader :version
+  attr_reader :user, :last_change_for_user
 
-  def initialize(version)
-    @version = version
+  def initialize(user)
+    @user = user
   end
 
-  def call
-    return if version.nil?
-
-    case type_of_update
-    when 'New'
-      return unless after_user.relevant_to_computacenter?
-    when 'Change'
-      return unless after_user.relevant_to_computacenter?
-    when 'Remove'
-      return unless before_user.relevant_to_computacenter?
+  def generate!
+    @last_change_for_user = Computacenter::UserChange.last_for(user)
```

Comment:
small non-blocking thing. can this be extracted to a memoized private method

Reply:
done 👍